### DISCUSSION TITLE
apply_inverse_transform function inverts transformed image to different size instead of original size?

### DISCUSSION THREAD

**Discussion by ruby0101** [TorchIO-project/torchio#493] [Q&A] [answered]
Hi, I have tried to invert a transformed image to the original size with apply_inverse_transform function. But after I applied the function, size is different than the original size as shown the below screenshot. How do I apply the inverse of all applied transforms and obtain my original image size?
<img width="723" alt="Screen Shot 2021-03-29 at 9 01 49 PM" src="https://user-images.githubusercontent.com/72773297/112841099-aa528a00-90d2-11eb-8898-5fa41f9a4a02.png">

**Comment by fepegar** [comment, marked as answer]
You cannot. At the moment, `Resample` is not invertible. Why do you want to apply the inverse transform?

**Reply by ruby0101** [reply]
I want to change a predicted label image to original image size. The predicted image size is (1, 48, 60, 48), original image size is (1, 512, 512, 86). Could I just apply resize transformation to it with no loss? Or how can I put back the predicted image to original size?
<img width="728" alt="Screen Shot 2021-03-29 at 9 34 19 PM" src="https://user-images.githubusercontent.com/72773297/112844930-d96afa80-90d6-11eb-8941-8351e1d11cd1.png">

**Comment by fepegar** [comment]
You can use `tio.Resample(original_image)(prediction)`.

However, I recommend you train your images in the original resolution. Images in the tutorial are downsampled so it can be run faster. You can use [patch-based training](https://torchio.readthedocs.io/data/patch_based.html) for that.

**Reply by ruby0101** [reply]
Thanks a lot, ok I will try to use patch-based training. I have faced an another issue with saving the predicted image as nii.gz format. Original image file size is only 7.6mb but when I save the predicted image, its size is so large almost 5 times greater than original image, it is 41mb. Do you know what is a problem with it? How could I save it with original image file size?
Here is my code to save the image:
<img width="584" alt="Screen Shot 2021-03-29 at 11 11 14 PM" src="https://user-images.githubusercontent.com/72773297/112862138-0a076000-90e8-11eb-8fa9-6f0cf8b4a8da.png">

**Reply by fepegar** [reply]
This is probably because the original and transformed images have different [data types](https://numpy.org/devdocs/user/basics.types.html).

**Reply by ruby0101** [reply]
Thank you so much, you are right, data types are different. Is there any available function in torchio to convert data types? How could I change data types of NIfTI images (int16 to float32)?
<img width="375" alt="Screen Shot 2021-03-30 at 3 18 50 AM" src="https://user-images.githubusercontent.com/72773297/112888875-79d91300-9107-11eb-84d4-77e2923b34f6.png">

**Reply by fepegar** [reply]
You can use the `data` property and `set_data` method of the `Image` class to modify the stored tensor data, e.g. `image.set_data(image.data.byte())`.

**Reply by ruby0101** [reply]
Thanks a lot, I have another issue. I have tried patch-based training but prediction result as follows:
<img width="743" alt="Screen Shot 2021-03-30 at 9 27 28 PM" src="https://user-images.githubusercontent.com/72773297/112996738-04208600-919f-11eb-83a5-1cd79ee60f3a.png">
What is the problem with my code? Could you tell me please what transformation should be applied for patch-based training? I applied the following transformation.
<img width="660" alt="Screen Shot 2021-03-30 at 9 27 55 PM" src="https://user-images.githubusercontent.com/72773297/112997078-5d88b500-919f-11eb-954c-931c10a19bf4.png">

**Reply by fepegar** [reply]
There are many potential causes of poor results. I suggest you follow Andrej Karpathy's advice: http://karpathy.github.io/2019/04/25/recipe/

I can help you with questions related to TorchIO, but I'm unfortunately not available to supervise all steps of your project, sorry!

**Reply by ruby0101** [reply]
Thank you so much for your all support. You can close this Q&A as answered status. Thanks.
